docs/architecture.md

```diff
@@ -0,0 +1,122 @@
+
+The following represents a non exhaustive list of topics relevant to be discussed when designing the 10101 architecture and is primarily based on the assumed component diagram introduced in https://github.com/get10101/10101/pull/30. Please also note that the order does not represent a priority of topics.
+
+## Database
+
+Both app and coordinator need to persist data. This section addresses the corresponding database decisions.
+
+### What data needs to be stored?
+
+ - Trading relevant data, like Offers, Orders, etc.
+ - Lightning channel related data.
+ - DLC related data - lightning channel extension data.
+
+Assumption: The app and coordinator hold mostly the same kind of data.
+
+### Should we use the same database / schemas for the app and coordinator?
+
+- **Option 1: Use the *same database***: The data relevant to be stored is structurally very similar for the app and the coordinator. In the poc (as well as in itchysats) we've used the same database abd schemas for both databases (taker and maker).
+  - **Advantage**: Maintaining migrations as well as the data access layer code is simplified.
+
+- **Option 2: Use *separate databases***: While using the same database makes sense from a maintenance point of view it might restrict us in the technology choices since a mobile app and backend service do have completely differnt constraints. It might as well hinder us from optimising our backend service as we have to consider mobile constraints.
+  - **Advantage**: Databases can be optimised according to the platform specific needs.
+
+### What kind of database should be used?
+
+- **Option 1: Use a *relational database***: In the past we have used relational databases, promoting a clear data structure and model. A lot of care has been taken to migrate old records to adhere to data model changes. No data redundancy.
+- **Option 2: Use a *no sql database***: A much more flexible option in terms of data model, with performance prioritised over data integrity. Data redundancy.
+
+
+## Lightning node
+
+Our design principles promote self-custody and self-sovereignty - as a result the users lightning node must run on her phone.
+
+### How should the lightning node run on the phone?
+
+- **Option 1: *Foreground process***: The lightning node will be started only when the app is opened.
+  - **Advantages**: low battery consumption, and simple implementation
+- **Option 2: *Background process***: The lightning node runs as background task on the phone, even if the app is closed.
+  - **Advantages**: payments can be received at any time
+
+Another alternative has been discussed if push notifications could be used to notify the user about an incoming payment asking to accept it by opening the app. We have not addressed the question of privacy.
+
+However, this feature might not be important for the MVP.
```

Comment:
Agreed, this feature is not needed for the MVP and an awesome feature to announce later on :)